.github/workflows/review_status.yml

```diff
@@ -0,0 +1,124 @@
+        # In the project we have each reviewers name listed in 
+        # A single Select option 3 times, one in each of 3 colors
+        # Indicating approval (green), changes needed (red), or
+        # review pending (gray).  Later on, we use this color
+        # along with the reviewer identity to fetch the 
+        # Single select id for that user in the appropriate color
+        case "$ACTION" in
+        "approved")
+          echo "ACTION_COLOR=GREEN" >> $GITHUB_ENV
+          ;;
+        "changes_requested")
+          echo "ACTION_COLOR=RED" >> $GITHUB_ENV
+          ;;
+        "dismissed")
+          echo "ACTION_COLOR=GRAY" >> $GITHUB_ENV
+          ;;
+        esac
+    - name: "Identify associated issues"
+      run: |
+        # Ok, to start with here, we use graphql to fetch the Referenced
+        # issues associated with a given PR.  These are the issues linked in the
+        # Development pane on the right side of the PR.  Record those issues to
+        # the file issues.json
+        gh api graphql -f query="query { repository(owner:\"openssl\", name:\"openssl\") { pullRequest(number: ${PR_NUM}) { closingIssuesReferences(first: 10) { edges { node { number } } } } } }" > ./issues.json
+    - name: "Find and update issues in project board"
+      run: |
+        # Now, for each issue we find above, we have some work to do
+        for i in $(jq '.data.repository.pullRequest.closingIssuesReferences.edges[].node.number' ./issues.json)
+        do
+          echo " ISSUE $i "
+          # For each issue, use some graphql to interrogate it to see if it
+          # A) Has an associated issue, and if so
+          # B) If that Issue exists on a project board which has a field
+          # named either "Review 1" or "Review 2".  If it does, 
+          # output the project, field and item id, along with the single select field
+          # optinos for the reviewer submitting this review.  Output that data
+          # to the files review1.json and review2.json
+          gh api -F issue=$i -F reviewer="$REVIEWER" graphql -f query='query($issue: Int!, $reviewer: [String!]) { repository(owner:"openssl", name:"openssl") { issue(number: $issue) { projectItems(first: 50) { edges { node { id project {title id field(name:"Review 1") { ... on ProjectV2SingleSelectField { id options(names: $reviewer) { id color} } } } fieldValueByName(name: "Review 1") { ... on ProjectV2ItemFieldSingleSelectValue { status: name color } } } } } } } }' > ./review1.json
+
+          gh api -F issue=$i -F reviewer="$REVIEWER" graphql -f query='query($issue: Int!, $reviewer: [String!]) { repository(owner:"openssl", name:"openssl") { issue(number: $issue) { projectItems(first: 51) { edges { node { id project {title id field(name:"Review 2") { ... on ProjectV2SingleSelectField { id options(names: $reviewer) { id color} } } } fieldValueByName(name: "Review 2") { ... on ProjectV2ItemFieldSingleSelectValue { status: name color } } } } } } } }' > ./review2.json
```

Comment:
`projectItems(first: 50)` in one, `projectItems(first: 51)` on the other.  Is that on purpose?

Reply:
No, thats just sloppiness on my part.  the first parameter fetches the first X linked projects to the issue.  Since we don't have but 9 projects open, the difference between 50 and 51 is irrelevant, but I'll bring them into line